source/appModules/soffice.py

```diff
@@ -31,20 +31,24 @@
 
 
 class SymphonyTextInfo(IA2TextTextInfo):
-
-	def _getFormatFieldAndOffsets(self,offset,formatConfig,calculateOffsets=True):
-		obj = self.obj
-		try:
-			startOffset,endOffset,attribsString=obj.IAccessibleTextObject.attributes(offset)
-		except COMError:
-			log.debugWarning("could not get attributes",exc_info=True)
-			return textInfos.FormatField(),(self._startOffset,self._endOffset)
+	def _getFormatFieldFromLegacyAttributesString(  # noqa: C901
+			self, attribsString: str,
+			offset: int) -> textInfos.FormatField:
+		"""Get format field with information retrieved from a text
+		attributes string containing LibreOffice's legacy custom text
+		attributes (used by LibreOffice <= 7.6), instead of attributes
+		according to the IAccessible2 text attributes specification
+		(used by LibreOffice >= 24.2).
+
+		:param attribsString: Legacy text attributes string.
+		:type attribsString: str
+		:param offset: Character offset for which the attributes are
+					   retrieved.
+		:type offset: int
+		:return: Format field containing the text attribute information.
+		:rtype: textInfos.FormatField

```

Comment:
We [recently updated our docstrings guide](https://github.com/nvaccess/nvda/blob/master/projectDocs/dev/codingStandards.md#docstrings).
Providing type information in docstrings is discouraged, instead we rely on python's type annotations.

```suggestion
		:param attribsString: Legacy text attributes string.
		:param offset: Character offset for which the attributes are
					   retrieved.
		:return: Format field containing the text attribute information.
```

Reply:
Done in new version.